### ISSUE TITLE
when md to latex with --listings, how to insert `$`

### ISSUE THREAD

**Issue by bubifengyun** [jgm/pandoc#4713]
when translate some code in lstinline, I find that

<pre>
\passthrough{\lstine!xxx!}
</pre>

I want to insert $, I know in latex should be 

<pre>
lstinline !$\$$!
</pre>

however, I used in markdown, I input like

<pre>
`$` => lstinline !$!
</pre>

<pre>
`$\$$` => lstinline !$\\$$!
</pre>

anyone knows how to insert $ in markdown? Thank you.

**Comment by mb21** [comment]
Thank you for submitting a bug report. However, we cannot improve pandoc or help you unless you give us all of the following information:

* the pandoc version (check using `pandoc -v` and try to reproduce the bug with the latest released version of pandoc, or even better: the development version or [nightly build](https://github.com/pandoc-extras/pandoc-nightly).)
* the exact command line used
* the exact input used
* the output received
* the output you expected instead

A small test case (just a few lines) is ideal.  If your input is large, try to whittle it down to a **minimum working example**.

Finally, ask questions on [pandoc-discuss](http://groups.google.com/group/pandoc-discuss) and read the [User's Guide](http://pandoc.org/MANUAL.html).

**Comment by bubifengyun** [comment]
ok, thank you.

**Comment by bubifengyun** [comment]
https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Writers/LaTeX.hs

I find a bug.

https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Writers/LaTeX.hs?#L1048#L1053

when set `mathescape=true`, lstinline cannot treat string between two `$` very well,
such as

<pre>
`$x_w$`
`$\$$`
</pre>

I want to reissue a bug report.

**Comment by mb21** [comment]
Thank you for submitting a bug report. However, we cannot improve pandoc or help you unless you give us all of the following information:

* the pandoc version (check using `pandoc -v` and try to reproduce the bug with the latest released version of pandoc, or even better: the development version or [nightly build](https://github.com/pandoc-extras/pandoc-nightly).)
* the exact command line used
* the exact input used
* the output received
* the output you expected instead

A small test case (just a few lines) is ideal.  If your input is large, try to whittle it down to a **minimum working example**. (For binary input formats like docx, you should attach the minimum sample file.)

Finally, ask questions on [pandoc-discuss](http://groups.google.com/group/pandoc-discuss) and read the [User's Guide](http://pandoc.org/MANUAL.html).

**Comment by bubifengyun** [comment]
Thank you,

## pandoc version

```
pandoc 2.2.1
Compiled with pandoc-types 1.17.4.2, texmath 0.11, skylighting 0.7.0.2
```

## exact command

```
$ pandoc test.md --to latex --from markdown --output result.tex --template template.tex --pdf-engine xelatex --listings
$ xelatex result.tex
```

## exact input
here  two files, one `test.md`, one `template.tex`

This is  test.md,
<pre>
\lstset{mathescape=true}
`` `$x_1$` `` => 
`$x_1$` but it should be $x_1$
`$` is a sign for non-root user.
</pre>

here is template.tex

<pre>
\documentclass{article}
\RequirePackage{etoolbox}
\RequirePackage[centering,a4paper,body={16cm,22cm}]{geometry} %设置版面
\RequirePackage{fancyhdr}
\RequirePackage{pageslts}
\RequirePackage[titles]{tocloft}
\RequirePackage{mathtools,amsthm,amsfonts,amssymb,bm}
\RequirePackage[defaultsups]{newtxtext}
\RequirePackage{newtxmath}
\RequirePackage[opentype]{sourcecodepro}
\RequirePackage{upgreek}
\RequirePackage{wasysym}
\RequirePackage{anyfontsize}
\RequirePackage{metalogo,doc}
\RequirePackage{array}
\RequirePackage{threeparttable}
\RequirePackage{dcolumn}
\RequirePackage{multirow}
\RequirePackage{booktabs}
\RequirePackage{graphicx}
\RequirePackage{caption}
\RequirePackage[list=off]{bicaption}
\RequirePackage{subcaption}
\RequirePackage[backend=biber,style=gb7714-2015]{biblatex}
\RequirePackage{xcolor}
\RequirePackage{listings}
\RequirePackage{textcomp}
\RequirePackage[xetex, bookmarksnumbered, colorlinks, urlcolor=black, linkcolor=black, citecolor=black, plainpages=false, pdfstartview=FitH]{hyperref}
\RequirePackage{longtable}
\RequirePackage[perpage, bottom]{footmisc}
\RequirePackage[inline]{enumitem}
\RequirePackage{pdfpages}
\RequirePackage{calc}
\RequirePackage{algorithm, algorithmicx, algpseudocode}
\RequirePackage{siunitx}
\RequirePackage{tikz}
% https://github.com/jgm/pandoc/issues/4716
\newcommand{\passthrough}[1]{#1}


\begin{document}

$body$

\end{document}
</pre>

## the output received

the first command, nothing shows, get result.tex file.

The second command, get as follows.
<pre>
$ pandoc test.md --to latex --from markdown --output result.tex --template template.tex --pdf-engine xelatex --listings

$ xelatex result.tex 

This is XeTeX, Version 3.14159265-2.6-0.99999 (TeX Live 2018) (preloaded format=xelatex)
 restricted \write18 enabled.
entering extended mode
(./result.tex
LaTeX2e <2018-04-01> patch level 2
Babel <3.18> and hyphenation patterns for 84 language(s) loaded.
(/usr/local/texlive/2018/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/local/texlive/2018/texmf-dist/tex/latex/base/size10.clo))
(/usr/local/texlive/2018/texmf-dist/tex/latex/etoolbox/etoolbox.sty)
(/usr/local/texlive/2018/texmf-dist/tex/latex/geometry/geometry.sty
<skip many code>
(/usr/local/texlive/2018/texmf-dist/tex/latex/wasysym/uwasy.fd))
Runaway argument?
 \@empty \expandafter \@gobbletwo \fi \expandafter \lst@next \else \lst@InsideC
onvert@ \ETC.
! File ended while scanning use of \lst@InsideConvert@ey.
<inserted text> 
                \par 
<*> result.tex
              
? 
</pre>

## the output I expected instead

![_ _20180628230946](https://user-images.githubusercontent.com/4437214/42043177-64e91d48-7b28-11e8-97d6-e47e656446b2.png)

but it pops error. and nothing can work as I expected.

**Comment by mb21** [comment]
Well, the example is not exactly minimal... I cannot debug your latex template for you. But maybe we're getting there... What's wrong with the following output?

    echo '`$x_1$`' | pandoc --to latex --listings
    \passthrough{\lstinline!$x\_1$!}

**Comment by bubifengyun** [comment]
Yes, pandoc: md => latex will not pop error, but the latex is not expected. the latex will pops error in xelatex. It need modify by hand.

<pre>
echo '\lstset{mathescape=true} `$x_1$`, `$`' | pandoc --to latex --listings
</pre>

will get 

<pre>
\lstset{mathescape=true} \passthrough{\lstinline!$x\_1$!},
\passthrough{\lstinline!$!}
</pre>

I need modify it by hand to this latex

<pre>
\lstset{mathescape=true} \passthrough{\lstinline!$x_1$!},
\passthrough{\lstinline!$\$$!}
</pre>

Or, xelatx will pops error like 

<pre>
Runaway argument?
 \@empty \expandafter \@gobbletwo \fi \expandafter \lst@next \else \lst@InsideC
onvert@ \ETC.
! File ended while scanning use of \lst@InsideConvert@ey.
 
                \par 
</pre>
